### DISCUSSION TITLE
Reading binlog order of each transaction

### DISCUSSION THREAD

**Discussion by timozhang** [zendesk/maxwell#2091] [Q&A]
I have a case that inserts data into multiple tables in one transaction. 
For example:
In code, I do insert as bellow order:
insert table A;
insert table B;
insert table C;

It's the same order in bin-log, but looks like Maxwell read bin-log in reversed order. Reading it as C, B then A.

May I ask why?
Is there any setting to make it read as A, B then C?

**Comment by osheroff** [comment]
very odd.  what producer are you using?

**Reply by timozhang** [reply]
I'm using RabbitMQ.

**Reply by osheroff** [reply]
so it's likely RabbitMQ that's re-ordering things.  If you switch to the `stdout` producer, what order do you see then?

The other option is that the transaction is getting re-ordered somehow but I've never seen that before.

Maxwell just reads the binlogs in order and outputs them in order.

**Reply by timozhang** [reply]
Ok, thx.

**Reply by timozhang** [reply]
You're right. 
The order from Maxwell is right. 
Just somehow reordered by RabbitMQ.
